### ISSUE TITLE
retrieve job logs retry delays not inherited from localhost configuration

### ISSUE THREAD

**Issue by MartinDix** [cylc/cylc-flow#3507]
**Describe the bug**
With cylc 7.7.2 and earlier, the retrieve job logs retry delays setting for extra hosts was inherited from the localhost setting. From 7.8.0 to 7.8.4 this no longer happens. Other settings still seem to work as expected.

**Release version(s) and/or repository branch(es) affected?**
7.8.0 - 7.8,4

**Steps to reproduce the bug**

Our configuration has
```
[hosts]
    [[localhost]]
        use login shell = False
        ssh command = ssh -Y -oBatchMode=yes
        retrieve job logs = True
        retrieve job logs retry delays = PT10S, PT30S, PT1M, PT3M
        retrieve job logs max size = 2M
        retrieve job logs command = rsync -a --chmod=Du=rwx,Dgo=rx,Fu=rw,Fgo=r
    [[gadi.*]]
        task communication method = ssh
        cylc executable = /projects/access/bin/cylc
```
With 7.7.2, cylc get-global-config
```
[hosts]
    [[localhost]]
...
        retrieve job logs command = rsync -a --chmod=Du=rwx,Dgo=rx,Fu=rw,Fgo=r
        retrieve job logs max size = 2M
        retrieve job logs retry delays = PT10S, PT30S, PT1M, PT3M
...
    [[gadi.*]]
...
        retrieve job logs command = rsync -a --chmod=Du=rwx,Dgo=rx,Fu=rw,Fgo=r
        retrieve job logs max size = 2M
        retrieve job logs retry delays = PT10S, PT30S, PT1M, PT3M
...
```
With 7.8.0 and later
```
[hosts]
    [[localhost]]
...
        retrieve job logs command = rsync -a --chmod=Du=rwx,Dgo=rx,Fu=rw,Fgo=r
        retrieve job logs max size = 2M
        retrieve job logs retry delays = PT10S, PT30S, PT1M, PT3M
...
    [[gadi.*]]
...
        retrieve job logs command = rsync -a --chmod=Du=rwx,Dgo=rx,Fu=rw,Fgo=r
        retrieve job logs max size = 2M
        retrieve job logs retry delays = 
...
```

**Expected behavior**
Current version should set retrieve job logs retry delays for all hosts as happens for other settings

**Screenshots**

**Additional context**
Possibly related to #3426?

Now that we've found this it can be easily worked around so low priority.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by dpmatthews** [comment]
I'm not quite sure what configuring remote log retrieval for `localhost` would do. Presumably the only reason to configure it is as a a default for other hosts?

**Comment by hjoliver** [comment]
> Presumably the only reason to configure it is as a a default for other hosts?

Presumably yes..

**Comment by hjoliver** [comment]
Reproduced :bug:

**Comment by hjoliver** [comment]
List items are not being inherited from localhost.  It looks like we started defaulting to empty list instead of `None`, which screws up the inheritance logic. (Empty list looks like "item set to nothing" rather than "item not set").

**Comment by hjoliver** [comment]
The unit test now passes on Travis. :relieved:

**Comment by oliver-sanders** [comment]
Closed by https://github.com/cylc/cylc-flow/pull/3508
